primitives/src/test_data.rs

```diff
@@ -0,0 +1,1662 @@
+                        .try_into()
+                        .expect("hash is always 32 bytes"),
+                    piece_size: *sector.padded_piece_size,
+                    client: client.0.clone(),
+                    provider: self.storage_provider().account_id,
+                    label: BoundedVec::try_from(label).unwrap(),
+                    start_block: 100.into(),
+                    end_block: min_dur + 100.into(),
+                    storage_price_per_block: 5u32.into(),
+                    provider_collateral: 25u32.into(),
+                    state: DealState::Published,
+                };
+                sign_proposal::<T>(client.1.clone(), proposal)
+            })
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap_or_else(|_| panic!("deals did not fit into BoundedVec"))
+    }
+
+    pub fn pre_commit_sectors(
+        &self,
+        limit: u32,
+    ) -> BoundedVec<SectorPreCommitInfo<BlockNumberFor<T>>, ConstU32<MAX_SECTORS_PER_CALL>>
+    where
+        T: crate::configs::StorageProviderProvider,
+        BlockNumberFor<T>: From<u64>,
+    {
+        self.sectors
+            .iter()
+            .take(limit as usize)
+            .map(|sector| SectorPreCommitInfo {
+                seal_proof: self.seal_proof,
+                sector_number: sector.sector_number,
+                sealed_cid: sector.comm_r.cid().to_bytes().try_into().unwrap(),
+                deal_ids: vec![u64::from(sector.sector_number) as u64]
+                    .try_into()
+                    .unwrap(),
+                expiration: min(
+                    T::SectorMaximumLifetime::get(),
+                    T::MaxSectorExpiration::get(),
+                ),
+                unsealed_cid: sector.comm_d.cid().to_bytes().try_into().unwrap(),
+                seal_randomness_height: self.seal_randomness_height.into(),
+            })
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap()
+    }
+
+    pub fn prove_commit_sectors(
+        &self,
+        limit: u32,
+    ) -> BoundedVec<ProveCommitSector, ConstU32<MAX_SECTORS_PER_CALL>> {
+        self.sectors
+            .iter()
+            .take(limit as usize)
+            .map(|sector| {
+                let proofs: BoundedVec<_, ConstU32<MAX_POREP_PROOFS_PER_BLOCK>> = sector
+                    .proof
+                    .chunks_exact(MAX_SEAL_PROOF_BYTES as usize)
+                    .map(|chunk| chunk.to_vec().try_into().unwrap())
+                    .collect::<Vec<_>>()
+                    .try_into()
+                    .unwrap();
+                ProveCommitSector {
+                    sector_number: sector.sector_number,
+                    proofs,
+                }
+            })
+            .collect::<Vec<_>>()
+            .try_into()
+            .unwrap()
+    }
+
+    pub fn load(sector_size: SectorSize) -> BenchmarkData<T> {
+        match sector_size {
+            // DO NOT MODIFY
```

Comment:
Btw. shouldn't we require only one sector_size here to be loaded?
Like if we need benchmarks only for 1GiB/8MiB to not pollute the binary?

Reply:
I guess it was bothering me that we'll only use SectorSize 1GiB for benchmarks and have multiple sector sizes in this switch which are not necessary.